pkg/configfilearg/defaultparser.go

```diff
@@ -1,18 +1,22 @@
 package configfilearg
 
 import (
+	"github.com/rancher/k3s/pkg/cli/cmds"
 	"github.com/rancher/k3s/pkg/version"
 	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
 )
 
+var defaultParser = &Parser{
+	After:         []string{"server", "agent", "etcd-snapshot:1"},
+	FlagNames:     []string{"--config", "-c"},
+	EnvName:       version.ProgramUpper + "_CONFIG_FILE",
+	DefaultConfig: "/etc/rancher/" + version.Program + "/config.yaml",
+	ValidFlags:    map[string][]cli.Flag{"server": cmds.ServerFlags, "etcd-snapshot": cmds.EtcdSnapshotFlags},
```

Comment:
do we need to handle `agent` in here too? Not sure if it's necessary but it feels like an omission.

Reply:
I can expand it to handle agent, I was focused on server and etcd-snapshot because of the bug

Reply:
Nothing happens if we have agent, it just returns without trying to skip any arguments.